pkg/controller/ipsec/ipsec_controller.go

```diff
@@ -0,0 +1,185 @@
+package ipsec
+
+import (
+	"context"
+	c "crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/openshift/cluster-network-operator/pkg/controller/statusmanager"
+	"github.com/openshift/library-go/pkg/crypto"
+	csrv1 "k8s.io/api/certificates/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const ipsecSignerName = "network.openshift.io/ipsec"
+
+var ca *crypto.TLSCertificateConfig
+
+// Add and Start it when the Manager is Started.
+func Add(mgr manager.Manager, status *statusmanager.StatusManager) error {
+	return add(mgr, newReconciler(mgr, status))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager) reconcile.Reconciler {
+	return &ReconcileCSR{client: mgr.GetClient(), scheme: mgr.GetScheme(), status: status}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("ipsec-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to CetificateSigningRequest resource
+	err = c.Watch(&source.Kind{Type: &csrv1.CertificateSigningRequest{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Make an ephemeral (in-memory) CA.
+	// TODO: This is temporary as it should use the Operator PKI ovn-ca secret
```

Comment:
I think it would be better for ipsec to use its own CA, and just store that in a Secret.

Reply:
No problem with that but what is your thought process. I thought it would be better to have a centralized ovn-ca that could be reused? If I dont do this, I guess I can create another cert-rotation-controller like you have done.

Reply:
@squeed I could use the operator PKI to create a new CA. E.g. not instantiate another cert rotatation controller.

Reply:
Basically, the ovn-ca is really intended to only be used for ovn database communications; it would be better that ipsec uses a different ca hierarchy, since they're separate.

If all you need is a CA, then the cert-rotation-controller isn't necessary; it's only a few lines of code to create the Secret with the CA. Create, like, a 100-year CA and just be done with it.

Reply:
I think I should rotate the CA. I will use the operatorpki resource to create another certrotation controller and call it signer-ca or something like that. Hopefully I will have a more complete PR by the end of the week and you can review.

Reply:
> I think I should rotate the CA.

Can ipsec handle that case? Remember, rotating the CA is not atomic. linux ipsec will need to support multiple CAs at once.

Reply:
Good question but I am more worried about OVS which may not. I am not sure but probably something we will need to eventually do. For the moment, I have integrated with operator pki but can change in a future rework if required.